.github/workflows/luacheck.yml

```diff
@@ -12,11 +16,11 @@ env:
 
 jobs:
   luacheck:
-    # We want to run on external PRs, but not on our own internal PRs
-    # as they'll be run by the push to the branch.
-    if: ( github.event_name == 'push' ||
-        github.event.pull_request.head.repo.full_name != github.repository ) &&
-        ! endsWith(github.ref, '-notest')
+    # Run on pull request only if the 'notest' label is unset and this is
+    # an external PR (internal PRs trigger a run on push).
+    if: github.event_name != 'pull_request' ||
+        ( ! contains(github.event.pull_request.labels.*.name, 'notest') &&
+          github.event.pull_request.head.repo.full_name != github.repository )
```

Comment:
How it behaves in a fork? Will I see a job that stucks forever waiting for a runner?

Reply:
Quoting the [GitHub documentation](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories-1):

> Workflows don't run on forked repositories by default. You must enable GitHub Actions in the Actions tab of the forked repository.

We already run most our workflows on our own runners and don't care to disable them for forks so I think we can drop this check (I mentioned it in the commit message BTW):

https://github.com/tarantool/tarantool/blob/11e87877df9001a4972019328592d79d55d1bb01/.github/workflows/centos_8.yml#L10-L18

Reply:
Okay, thanks.